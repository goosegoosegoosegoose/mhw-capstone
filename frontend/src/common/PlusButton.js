import React, { useContext } from "react";
import { Button } from "react-bootstrap";
import UserContext from "../auth/userContext";

const PlusButton = ({id, spacing, plus}) => {
  const currentUser = useContext(UserContext);

  const handlePlus = () => {
    plus(id)
  }

  if (!currentUser) {
    return (
      <Button className={spacing} variant="info" style={{color: "#ffffff"}} size="sm" onClick={handlePlus}>+1</Button>
    )
  }

  return (
    <>
      <Button className={spacing} variant="info" style={{color: "#ffffff"}} size="sm" onClick={handlePlus}>+1</Button>
    </>
  )
}

export default PlusButton;