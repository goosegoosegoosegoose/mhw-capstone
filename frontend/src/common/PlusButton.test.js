import { render } from '@testing-library/react';
import PlusButton from "./PlusButton";

test('smoke test', () => {
  render(      
    <PlusButton />
  );
});

test('snapshot', () => {
  const { asFragment } = render(<PlusButton />);
  expect(asFragment()).toMatchSnapshot();
});