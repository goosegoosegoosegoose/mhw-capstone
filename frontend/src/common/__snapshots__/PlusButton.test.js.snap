// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-info btn-sm"
    style="color: rgb(255, 255, 255);"
    type="button"
  >
    +1
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
